keras_cv/layers/object_detection/roi_align.py

```diff
@@ -0,0 +1,384 @@
+    and resizing it using the correspoding feature map of that level.
+
+    Args:
+      features: A dictionary with key as pyramid level and value as features. The
+        features are in shape of [batch_size, height_l, width_l, num_filters].
+      boxes: A 3-D Tensor of shape [batch_size, num_boxes, 4]. Each row represents
+        a box with [y1, x1, y2, x2] in un-normalized coordinates.
+      output_size: A scalar to indicate the output crop size.
+      sample_offset: a float number in [0, 1] indicates the subpixel sample offset
+        from grid point.
+
+    Returns:
+      A 5-D tensor representing feature crop of shape
+      [batch_size, num_boxes, output_size, output_size, num_filters].
+    """
+
+    with tf.name_scope("multilevel_crop_and_resize"):
+        levels = list(features.keys())
+        min_level = int(min(levels))
+        max_level = int(max(levels))
+        features_shape = tf.shape(features[min_level])
+        batch_size, max_feature_height, max_feature_width, num_filters = (
+            features_shape[0],
+            features_shape[1],
+            features_shape[2],
+            features_shape[3],
+        )
+
+        num_boxes = tf.shape(boxes)[1]
+
+        # Stack feature pyramid into a features_all of shape
+        # [batch_size, levels, height, width, num_filters].
+        features_all = []
+        feature_heights = []
+        feature_widths = []
+        for level in range(min_level, max_level + 1):
+            shape = features[level].get_shape().as_list()
+            feature_heights.append(shape[1])
+            feature_widths.append(shape[2])
+            # Concat tensor of [batch_size, height_l * width_l, num_filters] for each
+            # levels.
+            features_all.append(
+                tf.reshape(features[level], [batch_size, -1, num_filters])
+            )
+        features_r2 = tf.reshape(tf.concat(features_all, 1), [-1, num_filters])
+
+        # Calculate height_l * width_l for each level.
+        level_dim_sizes = [
+            feature_widths[i] * feature_heights[i] for i in range(len(feature_widths))
+        ]
+        # level_dim_offsets is accumulated sum of level_dim_size.
+        level_dim_offsets = [0]
+        for i in range(len(feature_widths) - 1):
+            level_dim_offsets.append(level_dim_offsets[i] + level_dim_sizes[i])
+        batch_dim_size = level_dim_offsets[-1] + level_dim_sizes[-1]
+        level_dim_offsets = tf.constant(level_dim_offsets, tf.int32)
+        height_dim_sizes = tf.constant(feature_widths, tf.int32)
+
+        # Assigns boxes to the right level.
+        box_width = boxes[:, :, 3] - boxes[:, :, 1]
+        box_height = boxes[:, :, 2] - boxes[:, :, 0]
+        areas_sqrt = tf.sqrt(
+            tf.cast(box_height, tf.float32) * tf.cast(box_width, tf.float32)
+        )
+
+        levels = tf.cast(
+            tf.math.floordiv(
+                tf.math.log(tf.math.divide_no_nan(areas_sqrt, 256.0)), tf.math.log(2.0)
```

Comment:
There are some many magic number here. 

Is it possible to convert those numbers from mix/max level?